contracts/hardhat.config.ts

```diff
@@ -26,6 +28,10 @@ export default {
     sources: "./src",
     tests: "./test",
   },
+  typechain: {
```

Comment:
Can we do this instead and remove the extra .gitignore entry?

```
  typechain: {
    outDir: "./artifacts/typechain",
    target: "ethers-v5",
  },
```

Reply:
I think it would be more consistent to leave it at the root level. We do that will all other build artifacts:
* abi/
* artifacts/
* cahce/
* solc/

Feels strange to me to nest the typechain build into the hardhat build folder. What if artifacts/ gets overwritten during compilation? I don't know that it does but I wouldn't be surprised.